### PR TITLE
[EXPERIMENTAL] tinyfk backend 

### PR DESCRIPTION
**Pull request by HiroIshida**
### feature
An experimental branch using [tinyfk](https://github.com/HiroIshida/tinyfk) as a backend of collision checker. Currently, the API of the tinyfk is still not fixed, so I want to keep it experimental.

Using tinyfk makes the solving speed aboud 50 times faster in `example/collision_free_trajectory.py` compared to my previous PR https://github.com/iory/scikit-robot/pull/181. (If `with_base=True` solving time is 0.20 sec, if `with_base=False` solving time is 0.04 sec)

### how to use 
```
git clone -b tinyfk https://github.com/HiroIshida/scikit-robot.git
cd scikit-robot 
pip install . --user
```

### dependency on libtinyxml
Because tinyfk is using a shared library dependency with `tinyxml` thus, to use tinyfk you must install the dependency via: 
```
sudo apt install libtinyxml-dev
```  
But most roboticist probably already has it as long as they use ROS.

### CODE REVIEW

File: skrobot/planner/collision_checker.py [hunk 1]
```diff
@@ -82,6 +82,7 @@ def add_collision_link(self, coll_link):
         self.coll_coords_list.extend(coords_list)
         self.coll_radius_list.extend([R] * len(sphere_list))
         self.n_feature = len(self.coll_coords_list)
+        return centers, coords_list, sphere_list, R
```
Comment (iory) [inline skrobot/planner/collision_checker.py:85]: Please add `Returns` at this function's docstrings.

File: skrobot/planner/tinyfk_sqp_based.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                               av_start,
+                               av_goal,
+                               joint_list,
+                               n_wp,
+                               safety_margin=1e-2,
+                               with_base=False,
+                               weights=None,
+                               initial_trajectory=None,
+                               slsqp_option=None
+                               ):
```
Comment (iory) [inline skrobot/planner/tinyfk_sqp_based.py:16]: Please add a docstring. It can be helpful for users.

**Review by iory — CHANGES_REQUESTED** [review]
It's a good function.
Please add docstrings for developers and users.

**Comment by HiroIshida** [comment]
@iory 
Let me give me some time to add documentation. I cannot decided the design completely, so is still to early to write the documentation. However, the as the example shows, the usage is completely same as the original one.

**Comment by iory** [comment]
Sure!

**Pull request closed without merging**